compose/foundation/foundation/src/uikitMain/kotlin/androidx/compose/foundation/cupertino/CupertinoOverscrollEffect.kt

```diff
@@ -62,12 +63,18 @@ private data class CupertinoOverscrollAvailableDelta(
 
 @OptIn(ExperimentalFoundationApi::class)
 class CupertinoOverscrollEffect(
-    /*
+    /**
      * Density to be taken into consideration during computations; Cupertino formulas use
      * DPs, and scroll machinery uses raw values.
      */
     private val density: Float,
-    layoutDirection: LayoutDirection
+    layoutDirection: LayoutDirection,
+
+    /**
+     * Some consumers of overscroll effect apply clip by themselves and some don't, thus this flag is needed
+     * to update our modifier chain and make the clipping correct in every case while avoiding redundancy
+     */
+    val applyClip: Boolean
```

Comment:
I suggest to move comments from constructor args above the constructor declaration.
https://github.com/JetBrains/compose-multiplatform-core/pull/860